### PR TITLE
UUID support for swap

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Fixes: #882

### CODE REVIEW

File: src/lib/plugin_apis/swap.api [hunk 1]
```diff
@@ -42,6 +42,7 @@ gboolean bd_swap_is_tech_avail (BDSwapTech tech, guint64 mode, GError **error);
  * bd_swap_mkswap:
  * @device: a device to create swap space on
  * @label: (nullable): a label for the swap space device
+ * @uuid: (nullable): UUID for the swap space device
```
Comment (tbzatek) [inline src/lib/plugin_apis/swap.api:46]: Looking at the `mkswap` manpage, three special values are accepted: `clear`, `random` and `time`. Does it make sense to document them here or just put a reference to the manpage? The UUID string often bears some specific format, but the usual `c1b9d5a2-f162-11cf-9ece-0020afc76f16` is being accepted here.

File: src/lib/plugin_apis/swap.api [hunk 2]
```diff
@@ -99,3 +100,15 @@ gboolean bd_swap_swapstatus (const gchar *device, GError **error);
  * Tech category: %BD_SWAP_TECH_SWAP-%BD_SWAP_TECH_MODE_SET_LABEL
  */
 gboolean bd_swap_set_label (const gchar *device, const gchar *label, GError **error);
+
+/**
+ * bd_swap_set_uuid:
```
Comment (tbzatek) [inline src/lib/plugin_apis/swap.api:105]: Needs to be added to `docs/libblockdev-sections.txt`?

**Review by tbzatek — APPROVED** [review]
Very nice, thank you! :heart_eyes:

File: src/lib/plugin_apis/swap.api (hunk 1 above)
Comment (vojtechtrefny) [reply src/lib/plugin_apis/swap.api:46]: I swa that too, I don't think we need to list it here, I don't really see the use case for the special options in `mkswap` (`random` is default and `clear` doesn't really make sense when creating a new swap). `swaplabel` doesn't support these.

File: src/lib/plugin_apis/swap.api (hunk 2 above)
Comment (vojtechtrefny) [reply src/lib/plugin_apis/swap.api:105]: Fixed.

File: src/lib/plugin_apis/swap.api (hunk 1 above)
Comment (tbzatek) [reply src/lib/plugin_apis/swap.api:46]: Fair enough, little chance of using these anyway I suppose.

File: src/plugins/swap.h [hunk 3]
```diff
@@ -40,10 +41,11 @@ void bd_swap_close (void);
 
 gboolean bd_swap_is_tech_avail (BDSwapTech tech, guint64 mode, GError **error);
 
-gboolean bd_swap_mkswap (const gchar *device, const gchar *label, const BDExtraArg **extra, GError **error);
+gboolean bd_swap_mkswap (const gchar *device, const gchar *label, const gchar *uuid, const BDExtraArg **extra, GError **error);
 gboolean bd_swap_swapon (const gchar *device, gint priority, GError **error);
 gboolean bd_swap_swapoff (const gchar *device, GError **error);
 gboolean bd_swap_swapstatus (const gchar *device, GError **error);
 gboolean bd_swap_set_label (const gchar *device, const gchar *label, GError **error);
+gboolean bd_seap_set_uuid (const gchar *device, const gchar *uuid, GError **error);
```
Comment (tbzatek) [inline src/plugins/swap.h:49]: Uhm, typo!

**Review by tbzatek — CHANGES_REQUESTED** [review]

**Review by tbzatek — APPROVED** [review]
Tested, works fine now, thanks

**Pull request merged**